lib/kino/process.ex

```diff
@@ -0,0 +1,251 @@
+  the supervision tree using Mermaid.js using the PID of the supervisor.
+
+      Kino.Process.generate_supervision_tree(supervisor_pid)
+
+  You can also render the supervisor by passing the name of the supervisor
+  if the supervisor was started with a name.
+
+      Kino.Process.generate_supervision_tree(ImportantSupervisor)
+
+  You can also change the direction of the rendering by calling
+  `Kino.Process.generate_supervision_tree/2` with the `:direction` option.
+
+      Kino.Process.generate_supervision_tree(ImportantSupervisor, direction: :lr)
+  """
+
+  alias Kino.Markdown
+
+  @type supervisor :: pid() | atom()
+
+  defmodule SupervisorGraphNode do
+    @moduledoc false
+
+    defstruct [:id, :pid, :type]
+
+    @doc false
+    def new(id, pid, type) do
+      %__MODULE__{
+        id: id,
+        pid: pid,
+        type: type
+      }
+    end
+
+    @doc false
+    def generate_node(%__MODULE__{id: id, pid: pid, type: type}) do
+      type =
+        if id == 0 do
+          :root
+        else
+          type
+        end
+
+      display =
+        pid
+        |> Process.info(:registered_name)
+        |> case do
+          {:registered_name, []} -> inspect(pid)
+          {:registered_name, name} -> inspect(name)
+        end
+
+      "#{id}(#{display}):::#{type}"
+    end
+  end
+
+  defmodule SupervisorGraphEdge do
+    @moduledoc false
+
+    defstruct [:node_1, :node_2, :relationship]
+
+    @doc false
+    def new(node_1, node_2, relationship) do
+      %__MODULE__{
+        node_1: node_1,
+        node_2: node_2,
+        relationship: relationship
+      }
+    end
+
+    @doc false
+    def generate_mermaid_entry(%__MODULE__{node_1: node_1, node_2: node_2, relationship: :link}) do
+      "#{SupervisorGraphNode.generate_node(node_1)} -..- #{SupervisorGraphNode.generate_node(node_2)}"
+    end
+
+    def generate_mermaid_entry(%__MODULE__{
+          node_1: node_1,
+          node_2: node_2,
+          relationship: :supervisor
+        }) do
+      "#{SupervisorGraphNode.generate_node(node_1)} ---> #{SupervisorGraphNode.generate_node(node_2)}"
+    end
+  end
+
+  @doc """
+  Generates a Mermaid.js graph of a supervision tree. The provided supervisor can be either a named
+  process or a PID. The supervision tree is displayed with solid lines denoting supervisor-worker
+  relationships and dotted lines denoting links between processes. The graph rendering supports
+  the following options:
+
+  ## Options
+
+    * `:direction` - defines the direction of the Mermaid.js graph. The
+      value can either be `:td` (top-down) or `:lr` (left to right) with
+      the default being `:td`. Optional
+
+  """
+  @spec generate_supervision_tree(supervisor(), keyword()) :: Markdown.t()
+  def generate_supervision_tree(supervisor, opts \\ [])
+
+  def generate_supervision_tree(named_supervisor, opts) when is_atom(named_supervisor) do
+    case Process.whereis(named_supervisor) do
```

Comment:
Move the pid clause first and then do something like this:

```suggestion
  def generate_supervision_tree(named_supervisor, opts) do
    case GenServer.whereis(named_supervisor) do
```

This will also allow us to support via tuples. :)